app/models/update_log.rb

```diff
@@ -4,14 +4,42 @@ def self.setting_record
     Setting.find_or_create_by(key: 'metrics_update')
   end
 
-  def self.log_update(time)
+  def self.log_delay
     setting = self.setting_record
-    setting.value["last_update"] = time
+    constant_update =  setting.value['constant_update']
+    if constant_update 
+      seconds = constant_update.keys.length > 1 ? constant_update.map {|key, value| (Time.parse(constant_update[key]["end_time"].to_s) - Time.parse(constant_update[key-1]["end_time"].to_s) if key != constant_update.keys[0]).to_i} : [0]
+      setting.value['average_delay'] = seconds.inject(:+)/(seconds.length-1) if seconds.length > 1
+    end
     setting.save
   end
 
+  def self.log_updates(starttime, endtime)
+    setting = self.setting_record
+    update_record = setting.value['constant_update'] ? setting.value['constant_update'] : setting.value['constant_update'] = Hash.new
+    last_update = update_record.keys.last
+    max_updates = 10
```

Comment:
This is a constant, so instead of setting a variable within the method, you can define it within the class, outside the method, like `MAX_UPDATES_TO_KEEP = 10`.